### PR TITLE
Feature/18 deterministic grid generation

### PR DESCRIPTION
**Pull request by severin-prenitzer**
According to the issue (https://github.com/tum-ens/pylovo/issues/18), the grid generation (specifically the building clustering) is was not completely deterministic. This was fixed by addressing four different issues:

1) Non-deterministic k-means clustering: 
Previously, the k-means clustering was done by the PostGIS function ST_ClusterKMeans directly in the database. K-means randomly initializes the centroids and this function does not support setting a seed value to get reproducable results.

-> Solution: Now we read out the vertice IDs and coordinates for all buildings of a connected component (get_connected_component_geometries), cluster them with sklearn.cluster.KMeans and then write the cluster labels (kcids) back to the corresponding vertice IDs in the database (update_kmeans_cluster_multiple). The seed for the clustering is defined in the config_data.yaml (can be any value in the range but usually does not need to be changed). The function update_large_kmeans_cluster was removed.

2) Non-deterministic hierarchical clustering: 
The hierachical clustering used the function fcluster with the criterion "maxclust=2" to split a too large building cluster in two smaller ones. Unfortunately, this function does not always result in two clusters, for input clusters with few nodes it can also result in one output cluster, and this behavior is not even completely deterministic on the input.

-> Solution: We use the function cut_tree from the same library. It always returns the required number of clusters (2) when possible and is deterministic on the input.

3) Reordering in hierarchical clustering: 
While always creating the same cluster distribution, the hierarchical clustering still orders the clusters non-deterministically (e.g. two clusters switch bcids when running the generation again).

-> Solution: After creating all building clusters for a certain kcid, we reassign the bcids to the clusters based on the smallest vertice ID they contain with _order_clusters_by_min_vertice. This order is the same in every run.

4) Non-determinism on follow-up PLZs in generate_grid_for_multiple_plz: 
While generate_grid_for_single_plz behaves deterministic after applying the first three changes, for generate_grid_for_multiple_plz the results are only the same for the first PLZ in the list and the following PLZ's clusters deviate between different runs.

-> By deleting and re-creating the temporary database tables every time a new PLZ is processed instead of just truncating them, generate_grid_for_multiple_plz behaves like generate_grid_for_single_plz and is now also deterministic.

### CODE REVIEW

**Comment by severin-prenitzer** [comment]
We now use DROP IF EXISTS and CREATE on the temp tables to reset them and we also may want to execute a DROP IF EXISTS before every CREATE of a temp table anyway in case the grid generation stopped unexpectedly in the previous run. 
Therefore the function create_tem_tables now drops the temp tables before creating them if they already exists and replaces the function reset_tables.

**Review by bereba — APPROVED** [review]
